tfhe/src/high_level_api/integers/types/base.rs

```diff
@@ -299,11 +281,10 @@ where
 impl<P, ClearType> FheDecrypt<ClearType> for GenericInteger<P>
 where
     P: IntegerParameter,
-    P::Id: RefKeyFromKeyChain<Key = crate::integer::ClientKey>,
     crate::integer::ClientKey: DecryptionKey<P::InnerCiphertext, ClearType>,
 {
     fn decrypt(&self, key: &ClientKey) -> ClearType {
-        let key = self.id.unwrapped_ref_key(key);
```

Comment:
What was the use of id here?

Reply:
Previously each data type had its own set of keys, FheBoolClentKey, FheUint2ClientKey

Ids was the mechanism that allowd to write generic function were the id was responsible of retrieving the 
correct key for the given type.

Now that there is only one key shared by all types, it's necessary to use the id to get the inner key.

Reply:
I guess you meant "not necessary" in your last sentence.
Then, is it still necessary to have `id`s now?

Reply:
> I guess you meant "not necessary" in your last sentence.
Indeed

> Then, is it still necessary to have `id`s now?

They are still somewhat useful as they are the zero sized typed stored in a GenericInteger<P> to have a difference 
between FheUin8,16,etc.

Reply:
Ok

Reply:
Like maybe instead of making `GenericInteger` generic over some parameters `P` (which has an associated Id),
make it generic over the `Id` directly, has now all types without exception shares parameters